### DISCUSSION TITLE
Language Design Proposal: .await Keyword like in Rust

### DISCUSSION THREAD

**Discussion by deeprobin** [dotnet/csharplang#5452] [Language Ideas] [answered]
I had the idea to implement a new await-pattern like rust language.

## Current
```cs
await MyAsyncMethod();
```

## Proposal
```cs
MyAsyncMethod().await;
```

## References
* [Rust Documentation: Keyword await](https://doc.rust-lang.org/std/keyword.await.html)
* [async/.await Primer](https://rust-lang.github.io/async-book/01_getting_started/04_async_await_primer.html)
* [Reddit: Rust is the only language that gets `await` syntax right](https://www.reddit.com/r/rust/comments/hnbz78/rust_is_the_only_language_that_gets_await_syntax/)

**Comment by deeprobin** [comment]
This also simplifies the handling of nullable tasks

For ex.
```cs
Task? myNullableTask = null;
myNullableTask?.await; // gets only awaited if myNullableTask is not null
```

See https://github.com/dotnet/csharplang/discussions/5325

**Comment by 333fred** [comment, marked as answer]
Duplicate of https://github.com/dotnet/csharplang/issues/4076, which was explicitly rejected by the LDM.
